Applications/Spire/Include/Spire/Ui/FlatButton.hpp

```diff
@@ -69,9 +79,16 @@ namespace Spire {
       //! Sets the style of the button when it's disabled.
       void set_disabled_style(const Style& disabled_style);
 
-      //! Connects a slot to the clicked signal.
-      boost::signals2::connection connect_clicked_signal(
-        const ClickedSignal::slot_type& slot) const;
+      //! Sets the pixel size and weight of the button font.
+      /*!
+        \param font_pixel_size The pixel size of the font.
+        \param font_weight The weight of the font.
+      */
+      void set_font_size_weight(int font_pixel_size, int font_weight);
```

Comment:
The set_font_properties() method proposed here is similar to the function setFont() which the qWidget class owns. In our cases, the font family remains the same, and only the font size and weight are variables, so this is the reason why I added this function and chose set_font_size_weight as its name.  I know this name is ugly, and we may replace it with setFont() like the previous version. The only downside is we have to set a font family every time when we want to change the font size or weight. Which one do you prefer?

Reply:
@kranar can weigh in, but set_font_properties() seems fine. The changeable properties are inherently documented because they're the parameters. My issue with implicitly reusing setFont() is that it makes it appear as if the entire font can be changed.

There's multiple cases in Spire where a base class method can be used instead of calling one of our Spire subclass methods. For instance, the IconButton has its own Style struct to set the button's style, but since it subclasses the QAbstractButton you could still technically call setStyleSheet(...) on it. I don't have a perfect solution, but as a rule of thumb the Spire class' API takes implicit precedence over the QWidget API, if there is a potential conflict.